azkaban-webserver/src/restli/java/azkaban/restli/ProjectManagerResource.java

```diff
@@ -97,22 +101,56 @@ public String deploy(@ActionParam("sessionId") String sessionId,
       // complete.
       logger.info("Downloading package from " + packageUrl);
       FileUtils.copyURLToFile(url, archiveFile);
-      Props props = new Props();
 
       logger.info("Downloaded to " + archiveFile.toString());
-      projectManager.uploadProject(project, archiveFile, "zip", user, props);
     } catch (IOException e) {
       String errorMsg =
           "Download of URL " + packageUrl + " to " + archiveFile.toString()
               + " failed";
       logger.error(errorMsg, e);
-      throw new ProjectManagerException(errorMsg, e);
+      if (tempDir.exists()) {
```

Comment:
It is not new logic. Previously, if the download or validation failed for some exception, we would delete the directory if it existed.
Now, I separated the downloading part and the validation part of the project file. If there is some kind of exception during the download, we delete the file (as done before). If the download succeeds, we try to validate it. Regardless of validation failing or not, the directory will need to be deleted, so there is another directory delete under the try clause trying to validate the file.